packaging/setup_win32.py.in

```diff
@@ -1,141 +1,88 @@
-copytree("@EXTERNALPROJECT_INSTALL_PREFIX@/share/uranium/resources", "dist/resources")
-copytree("@EXTERNALPROJECT_INSTALL_PREFIX@/share/cura/resources", "dist/resources")
-
-print("Coping Uranium QML.")
-copytree("@EXTERNALPROJECT_INSTALL_PREFIX@/lib/python3.4/site-packages/UM/Qt/qml/UM", "dist/qml/UM")
-
-qt_lib_dir = None
-qt_bin_dir = None
-if os.path.exists("@EXTERNALPROJECT_INSTALL_PREFIX@/lib/plugins"):
-    qt_lib_dir = "@EXTERNALPROJECT_INSTALL_PREFIX@/lib"
-    qt_bin_dir = "@EXTERNALPROJECT_INSTALL_PREFIX@/bin"
-else:
-    import PyQt5.QtCore
-    qt_lib_dir = os.path.dirname(PyQt5.QtCore.__file__)
-    qt_bin_dir = os.path.dirname(PyQt5.QtCore.__file__)
-
-print("Copying Qt5 plugins")
-copytree(os.path.join(qt_lib_dir, "plugins"), "dist/PyQt5/plugins")
-print("Copying QtQuick plugins")
-copytree(os.path.join(qt_lib_dir, "qml"), "dist/qml")
-print("Copying Qt5 svg library")
-shutil.copy(os.path.join(qt_bin_dir, "Qt5Svg.dll"), "dist/Qt5Svg.dll")
 
+from cx_Freeze import setup, Executable, hooks
+
+def load_scipy(finder, module):
+    finder.IncludePackage("scipy._lib")
+    finder.IncludePackage("scipy.misc")
+    finder.IncludePackage("scipy.sparse.csgraph._validation")
+
+hooks.load_scipy = load_scipy
+
+def load_pyqt5_qtquick(finder, module):
+    finder.IncludeModule("PyQt5.QtCore")
+    finder.IncludeModule("PyQt5.QtGui")
+    finder.IncludeModule("PyQt5.QtQml")
+    finder.IncludeModule("PyQt5.QtNetwork")
+    finder.IncludeModule("PyQt5._QOpenGLFunctions_2_0")
+
+hooks.load_PyQt5_QtQuick = load_pyqt5_qtquick
+
+search_path = sys.path.copy()
+search_path.insert(1, "@EXTERNALPROJECT_INSTALL_PREFIX@/lib/python3.5/site-packages/")
+search_path.insert(2, "@CMAKE_PREFIX_PATH@/lib/python3.5/site-packages/")
+
+# Dependencies are automatically detected, but it might need
+# fine tuning.
+build_options = {
+    "build_exe": "package",
+    "path": search_path,
+    "packages": [
+        "xml.etree",
+        "uuid",
+        "serial",
+        "zeroconf",
+        "UM",
+        "cura",
+        "stl",
+    ],
+    "include_files": [
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/bin/CuraEngine.exe", ""),
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/lib/cura", "plugins"),
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/lib/uranium", "plugins"),
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/lib/python3.5/site-packages/UM/Qt/qml/UM", "qml/UM"),
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/share/cura/resources", "resources"),
+        ("@EXTERNALPROJECT_INSTALL_PREFIX@/share/uranium/resources", "resources"),
+        # Preinstalled PyQt5 installation
+        (PyQt5.__path__[0] + "/Qt/qml/Qt", "qml/Qt"),
+        (PyQt5.__path__[0] + "/Qt/qml/QtQml", "qml/QtQml"),
+        (PyQt5.__path__[0] + "/Qt/qml/QtQuick", "qml/QtQuick"),
+        (PyQt5.__path__[0] + "/Qt/qml/QtQuick.2", "qml/QtQuick.2"),
+    ],
+    "excludes": [ ]
+}
+
+# Extra libraries for numpy.core*
 print("Copy numpy DLLs")
 numpy_core_dir = os.path.dirname(numpy.core.__file__)
```

Comment:
Is this still needed with cx_Freeze? As far as I know it has better handling of dependencies so we do not need code like this.